client/js/lounge.js

```diff
@@ -220,6 +220,33 @@ $(function() {
 	});
 
 	socket.on("init", function(data) {
+		if (!$("#loading").length) {
+			var channels = $.map(data.networks, function(n) {
+				return n.channels;
+			});
+
+			channels.forEach(function(channel) {
+				renderChannel(channel);
+				if (channel.unread > 0 && channel.type === "channel") {
+					var badge = sidebar.find(".chan[data-title='" + channel.name + "'] .badge");
+					badge.text(channel.unread);
+					if (channel.highlight) {
+						badge.addClass("highlight");
+					}
+				}
+			});
+
+			if (sidebar.find(".highlight").length) {
```

Comment:
This doesn't appear to update sidebar. If the channels or networks change while you are offline, this would be out of sync.